apm-agent-plugins/apm-log-shader-plugin/apm-log4j2-plugin/src/main/java/co/elastic/apm/agent/log4j2/Log4j2LogShadingHelper.java

```diff
@@ -95,6 +73,41 @@ protected String getAppenderName(AbstractOutputStreamAppender<?> appender) {
             .setStackTraceAsArray(false)
             .build();
 
+        Appender ret = null;
+
+        if (isOverrideConfigured()) {
```

Comment:
Is it an issue when the config is dynamically changed at runtime? The value returned by this method will still be the same cached value. For example, when changing from `override` to `replace`, the map will contain a `EcsLayoutHolder`.

Reply:
Yes, the dynamic handling is not sorted out yet, it is on my todo list, including proper testing. I proposed two approaches for that at the initial comment, would love to get your input on that